src/keyboard.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license Copyright (c) 2003-2016, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+import CKEditorError from './ckeditorerror.js';
+
+/**
+ * Set of utils related to keyboard support.
+ *
+ * @namespace utils.keyboard
+ */
+
+/**
+ * Object with `keyName => keyCode` pairs.
+ *
+ * Contains:
+ *
+ * * `a-z`,
+ * * `0-9`,
+ * * `arrow(left|up|right|bottom)`,
+ * * `backspace`, `delete`, `enter`,
+ * * `ctrl`, `cmd`, `shift`, `alt`.
+ *
+ * @member {Object} utils.keyboard.knownKeyNames
+ */
+export const knownKeyNames = generateKnownKeyNames();
+
+/**
+ * Converts a key name or a {@link utils.keyboard.KeystrokeInfo keystroke info} into a key code.
+ *
+ * Note: Key names are matched with {@link utils.keyboard.knownKeyNames} in a case-insensitive way.
+ *
+ * @method utils.keyboard.getCode
+ * @param {String|utils.keyboard.KeystrokeInfo} Key name (see {@link utils.keyboard.knownKeyNames})
+ * or a keystroke data object.
+ * @returns {Number} Key or keystroke code.
+ */
+export function getCode( key ) {
```

Comment:
An example would be useful. It is not obvious what the format of the `key` is using only a docs.